SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -485,11 +490,33 @@ bool FLocalDeploymentManager::GetServiceStatus()
 		bLocalDeploymentRunning = false;
 		return false;
 	}
-	else
+	else if (ServiceStatusResult.Contains(TEXT("Local API service is running")))
 	{
-		UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Spatial service running."));
-		bSpatialServiceRunning = true;
-		return true;
+		// Get the project file path and ensure it matches the one for the currently running project.
+		FString SpatialServicePath;
+		if (ServiceStatusResult.Split(TEXT("project file path: "), nullptr, &SpatialServicePath))
+		{
+			// Remove the trailing '" cli-version' that comes with non-interactive 'spatial' calls.
+			SpatialServicePath.Split(TEXT("\" cli-version"), &SpatialServicePath, nullptr);
+
+			FString CurrentProjectSpatialPath = FPaths::Combine(FSpatialGDKServicesModule::GetSpatialOSDirectory(), TEXT("spatialos.json"));
+			FPaths::NormalizeDirectoryName(SpatialServicePath);
+			FPaths::RemoveDuplicateSlashes(SpatialServicePath);
+			if (CurrentProjectSpatialPath == SpatialServicePath)
+			{
+				bSpatialServiceInProjectDirectory = true;
+			}
+			else
+			{
+				UE_LOG(LogSpatialDeploymentManager, Error,
+					TEXT("Spatial service running in a different project! Please run 'spatial service stop' if you wish to launch deployments in the current project. Service at: %s"), *SpatialServicePath);
+				bSpatialServiceInProjectDirectory = false;
+			}
+
+			UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Spatial service running at path: %s "), *SpatialServicePath);
+			bSpatialServiceRunning = true;
+			return true;
```

Comment:
Should GetServiceStatus() return true in case bSpatialServiceInProjectDirectory = false?

Reply:
Yup, done